content/unified_mode.md

```diff
@@ -12,4 +12,170 @@ product = ["client"]
+
+Using `unified_mode` this resource is simplified:
+
+```ruby
+unified_mode true
+
+provides :downloader
+
+action :doit do
+  remote_file "/tmp/users.json" do
+    source "https://jsonplaceholder.typicode.com/users"
+  end
+
+  array = FFI_Yajl::Parser.parse(IO.read("/tmp/users.json"))
+
+  file "/tmp/phone" do
+    content array.last["phone"]
+  end
+end
+```
+
+The need for the `ruby_block`, the `lazy` and the variable declaration are all removed, and all that is left is how a beginning programmer would attempt to write this resource without the need for deep Chef Infra or Ruby knowledge.
+
+### Recovery and Exception Handling
+
+Another advantage is in error recovery and the use of rescue:
+
+```ruby
+unified_mode true
+
+provides :redis_install
+
+action :install do
+  version = "5.0.5"
+
+  # the downloading of this file acts as a guard for all the later
+  # resources -- but if the download is successful while the later
+  # resources fail for some transient issue, will will not redownload on
+  # the next run -- we lose our edge trigger.
+  #
+  remote_file "/tmp/redis-#{version}.tar.gz" do
+    source "http://download.redis.io/releases/redis-#{version}.tar.gz"
+    notifies :run, "execute[unzip_redis_archive]", :immediately
+  end
+
+  begin
+    execute "unzip_redis_archive" do
+      command "tar xzf redis-#{version}.tar.gz"
+      cwd "/tmp"
+      action :nothing
+      notifies :run, "execute[build_and_install_redis]", :immediately
+    end
+
+    execute "build_and_install_redis" do
+      command 'make && make install'
+      cwd "/tmp/redis-#{version}"
+      action :nothing
+    end
+  rescue Exception
+    file "/tmp/redis-#{version}.tar.gz" do
+      action :delete
+    end
+    raise
+  end
+end
+```
+
+This simplified example shows how to trap exceptions from resources using normal ruby syntax and to clean up the resource. Without `unified_mode` this syntax is impossible. Normally the `execute` resources when they are parsed only creates the objects in the `resource_collection` to later be evaluated so that no exception is thrown while ruby is parsing the `action` block at all. Every action is delayed to the later converge phase. In `unified_mode` the resource runs when ruby is done parsing its block, so exceptions happen in-line with ruby parsing, so the `rescue` now works as naively expected.
+
+The usefulness of this is that if the tar extraction throws an exception (for example, it could be out of disk space), then that Exception causes the tar file to be deleted and then redownloaded the next time `chef-client` is run. Without the cleanup the tar file would exist, but the resource would not have completed, the extraction would not happen and the resource would be left in a broken, indeterminate state.
```

Comment:
```suggestion
The usefulness of this is that if the tar extraction throws an exception (for example, it could be out of disk space), then that Exception causes the tar file to be deleted and then redownloaded the next time Chef Infra Client is run. Without the cleanup the tar file would exist, but the resource would not have completed, the extraction would not happen and the resource would be left in a broken, indeterminate state.
```